api/src/service/subproject_permission_grant.ts

```diff
@@ -39,7 +40,6 @@ export async function grantSubprojectPermission(
     ),
   );
   if (Result.isErr(result)) return Promise.reject(result);
-
```

Comment:
above: get result, check result, return if result is error

below: store the events that got generated by the domain layer

why not keep them separated?